lib/awful/keyboard.lua

```diff
@@ -0,0 +1,77 @@
+---------------------------------------------------------------------------
+--- Utilities related to the keyboard and text entry.
+--
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2018 Emmanuel Lepage Vallee
+-- @module awful.keyboard
+---------------------------------------------------------------------------
+local capi = {root = root, client = client}
+
+local module = {}
+
+-- The hash order isn't consistent when the content changes, but it is
+-- constant when the table doesn't change.
+local conversion = {
+    Super_L   = "Mod4",
+    Control_L = "Control",
+    Shift_L   = "Shift",
+    Alt_L     = "Mod1",
+    Super_R   = "Mod4",
+    Control_R = "Control",
+    Shift_R   = "Shift",
+    Alt_R     = "Mod1",
+}
+
+--- Send an artificial set of key events to trigger a key combination.
+--
+-- This can be used to call application actions or Awesome own keybindings.
+--
+-- @tparam table modifiers The modifiers (like `Control` or `Mod4`).
+-- @tparam string|number key The key name or keycode.
+-- @function awful.keyboard.execute_keybindings
+function module.execute_keybinding(modifiers, key)
```

Comment:
I don't think this function works when some "unwanted modifier" is currently pressed. E.g., I won't be able to bind mod4+Y to press Alt+L, because it will actually do mod4+Alt+L.